init-tools.cmd

```diff
@@ -47,9 +47,32 @@ echo Running %0 > "%INIT_TOOLS_LOG%"
 set /p DOTNET_VERSION=< "%~dp0DotnetCLIVersion.txt"
 if exist "%DOTNET_CMD%" goto :afterdotnetrestore
 
+REM Use x86 tools on arm64 and x86.
+REM arm32 host is not currently supported, please crossbuild.
+if "%PROCESSOR_ARCHITECTURE%" == "ARM" (
+  echo "Error, arm32 arch not supported for build tools."
+  exit /b 1
+)
+
+if "%PROCESSOR_ARCHITECTURE%" == "AMD64" (
+  set _Arch="x64"
+  goto ArchSet
+)
+
+REM If this is not amd64 and not arm, then we should be running on arm64 or x86
+REM either way we can (and should) use the x86 dotnet cli
+REM
+REM TODO: consume native arm64 toolset, blocked by official arm64 windows cli
```

Comment:
Can you make sure there is a GitHub issue related to this, and reference the issue here?